kimchi/src/index.rs

```diff
@@ -172,8 +172,12 @@ pub fn constraints_expr<F: FftField + SquareRootField>(
             )
         }
     };
+
     if chacha {
-        expr + chacha::constraint(super::range::CHACHA.start)
+        let expr = expr + chacha::constraint_chacha0(super::range::CHACHA.start);
```

Comment:
Why split these? It was more readable before, and it would be easy to create a function that calls them all.

Reply:
the idea was that they are separate gates, so let's make that explicit